lightning/src/ln/peer_handler.rs

```diff
@@ -1036,11 +1038,13 @@ impl<Descriptor: SocketDescriptor, CM: Deref> PeerManager<Descriptor, CM> where
 
 	/// Indicates that the given socket descriptor's connection is now closed.
 	///
-	/// This must be called even if a PeerHandleError was given for a read_event or write_event,
-	/// but must NOT be called if a PeerHandleError was provided out of a new_\*\_connection event!
+	/// This must only be called if the socket has been disconnected by the peer or your own
+	/// decision to disconnect it and must NOT be called in any case where other parts of this
+	/// library (eg PeerHandleError, explicit disconnect_socket calls) instruct you to disconnect
```

Comment:
How is the user supposed to differentiate between their own call to `disconnect_socket` and that of this library?

Reply:
Hmm, `disconnect_socket` is different from `socket_disconnected` - I suppose a user *could* call `disconnect_socket` on their own Descriptor, but in general I presumed they wouldn't (and would just call shutdown(2)).